### DISCUSSION TITLE
how do I query /zwave/driver/status

### DISCUSSION THREAD

**Discussion by lxrootard** [zwave-js/zwave-js-ui#3619] [Q&A]
Hello

Many thanks for your great work on this lib.
I'm implementing a remote docker client that communicates through MQTT.
When I start `zwave-js-ui` after my code I get the `/zwave/driver/status` information, but when I start  `zwave-js-ui` before my code I'm not notified although I've set `RETAIN `to true.

Is there a way to refresh this information?
Many thanks

**Comment by robertsLando** [comment]
>  I'm not notified although I've set RETAIN to true.

This is exactly what retain is used for, in general MQTT shouldn't be used as REST API (request --> response) but it's event based (something happens --> notify subscribers). Retain is used so subscribers once subscribe to a specific topic receive the last message published to that topic so they can always keep ther state in sync

**Comment by lxrootard** [comment]
@robertsLando not sure to understand your answer? 
As I said I had set `RETAIN` to `true`. Am I missing something here or is this a bug?

The only alternative API I'm aware of is `/health` but correct me if I'm wrong there's no way to make a distinction between driver states?

**Comment by lxrootard** [comment]
Hello @robertsLando could you please advise?
Are there any options to check if the driver is ready using docker when the client connects after the lib's initial startup?
Thanks

**Comment by robertsLando** [comment]
@lxrootard What broker are you using? Does your broker stores retained messages on disk or on ram?

**Reply by lxrootard** [reply]
Thanks for your reply
I'm using mosquitto with persistence on disk.

I've finally managed to solve my issue by issuing an extra `zwave/driver/status/get` 
However I get no answer from `zwave/_CLIENTS/ZWAVE_GATEWAY-<mqtt_name>/version` ??

Still `zwave/_CLIENTS/ZWAVE_GATEWAY-<mqtt_name>/api/getInfo/set` works ok
Am I missing something?

**Reply by robertsLando** [reply]
> I've finally managed to solve my issue by issuing an extra zwave/driver/status/get. However I get no answer from zwave/_CLIENTS/ZWAVE_GATEWAY-<mqtt_name>/version ??

Sorry I don't undesrtand what you mean, I don't have any listener for such incoming message

> Still zwave/_CLIENTS/ZWAVE_GATEWAY-<mqtt_name>/api/getInfo/set works ok

I think this is the correct api you have to use in order to manually query the driver status. Anyway if you only need to know if driver is ready or not the `zwave/driver/status` receives a payload of `tree|false` based on that.

**Reply by lxrootard** [reply]
I meant App Version in [special topics](https://zwave-js.github.io/zwave-js-ui/#/guide/mqtt?id=app-version):
```
App version
<mqtt_prefix>/_CLIENTS/ZWAVE_GATEWAY-<mqtt_name>/version

The payload will be in the time-value json format and the value will contain the app string version.
```

**Reply by robertsLando** [reply]
@lxrootard Are you using mqtt explorer? Please try to use it to be sure you don't have any typo on subscriptions, that will show you all the tags in a very user friendly tree view of topics

**Reply by lxrootard** [reply]
yes I'm using MQTT explorer
I believe I misunderstood your doc eg this topic can be subscribed but it can't be queried right?

**Reply by robertsLando** [reply]
> I believe I misunderstood your doc eg this topic can be subscribed but it can't be queried right?

I don't undesrtand what you mean here, there is no way to 'query' a topic in mqtt
